pkg/daemon/ceph/osd/volume.go

```diff
@@ -1007,6 +1010,29 @@ func GetCephVolumeRawOSDs(context *clusterd.Context, clusterInfo *client.Cluster
 			Store:         "bluestore",
 		}
 
+		if !skipDeviceClass {
+			deviceProps, err := sys.GetDeviceProperties(blockPath, context.Executor)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to get device %q props", blockPath)
+			}
+
+			if deviceProps["ROTA"] == "1" {
+				osd.DeviceClass = "hdd"
+			} else {
+				// if used on PVC we don't know what is device class type - ssd or nvme, setting to ssd by default
+				devicePath := strings.Split(blockPath, "/")
+				if devicePath[0] == "dev" {
+					if strings.HasPrefix(devicePath[1], "nvme") {
```

Comment:
Instead of splitting on the device name, what if we just use `strings.Contains(blockPath, "nvme")`? An ssd very likely doesn't have nvme anywhere in the name, right? I'd vote for code simplicity over an extremely small corner case. It can also be overridden, correct?

Reply:
Based on leseb comment above related to using `RealPath` i guess that make sense yeah, i'll update.